File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.cpp
class Solution {
public:
    int minimumCardPickup(vector<int>& cards) {
        map <int, int> m;
        pair<int, int> minima ={-1, INT_MAX};
        for(int i=0; i<cards.size(); i++){
            if(m[cards[i]]){
                if(minima.second>i-m[cards[i]]+1)
                    minima = {cards[i], i-m[cards[i]]+1};
            }
            m[cards[i]]=i+1;
        }
        return minima.second==INT_MAX?-1:minima.second+1;
    }
};

File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.java
class Solution {
    public int minimumCardPickup(int[] cards) {
        int lt = 0, rt = 0, n = cards.length, ans = Integer.MAX_VALUE;
        Map<Integer, Integer> m = new HashMap<>();
        for(;rt<n; rt++){
            if(m.containsKey(cards[rt]))
                ans = Math.min(ans, rt+1-m.get(cards[rt]));
            m.put(cards[rt], rt);
        }
        return ans==Integer.MAX_VALUE?-1:ans;
    }
}